src/adapt/epub.js

```diff
@@ -109,34 +111,59 @@ adapt.epub.EPUBDocStore.prototype.startLoadingAsJSON = function(url) {
  * @return {!adapt.task.Result.<adapt.epub.OPFDoc>}
  */
 adapt.epub.EPUBDocStore.prototype.loadEPUBDoc = function(url, haveZipMetadata) {
-    const self = this;
+    let epubUrl = url;
     /** @type {!adapt.task.Frame.<adapt.epub.OPFDoc>} */ const frame
         = adapt.task.newFrame("loadEPUBDoc");
-    if (url.substring(url.length - 1) !== "/") {
-        url = `${url}/`;
+    if (epubUrl.substring(epubUrl.length - 1) !== "/") {
+        epubUrl = `${epubUrl}/`;
     }
     if (haveZipMetadata) {
-        self.startLoadingAsJSON(`${url}?r=list`);
+        this.startLoadingAsJSON(`${epubUrl}?r=list`);
     }
-    self.startLoadingAsPlainXML(`${url}META-INF/encryption.xml`);
-    const containerURL = `${url}META-INF/container.xml`;
-    self.loadAsPlainXML(containerURL, true, `Failed to fetch EPUB container.xml from ${containerURL}`).then(containerXML => {
-        if (!containerXML) {
-            vivliostyle.logging.logger.error(`Received an empty response for EPUB container.xml ${containerURL}. This may be caused by the server not allowing cross origin requests.`);
-        } else {
-            const roots = containerXML.doc().child("container").child("rootfiles")
-                .child("rootfile").attribute("full-path");
 
-            for (const root of roots) {
-                if (root) {
-                    self.loadOPF(url, root, haveZipMetadata).thenFinish(frame);
-                    return;
+    let r;
+    if (r = (/^(.*\/)([^/]+\.opf)$/).exec(url)) {
+        // EPUB OPF
+        epubUrl = r[1];
+        const root = r[2];
+        this.loadOPF(epubUrl, root, haveZipMetadata).thenFinish(frame);
+    } else if ((/\.json(?:ld)?$/).test(url)) {
```

Comment:
Just a thought: if the URL is `http[s]?`, you may also create a `HEAD` request to test for the `content-type` header (`application/webpub+json` || `application/audiobook+json` || `application/json`), otherwise you will miss URLs that do not use syntactical conventions like `manifest.json`.

Reply:
Fixed. Thanks! @danielweck